example/lib/custom_asset_loader.dart

```diff
@@ -22,10 +23,13 @@ class StringAssetLoader extends AssetLoader {
 //
 class FileAssetLoader extends AssetLoader {
   @override
-  Future<String> load(String localePath) {
+  Future<Map<String, dynamic>> load(String localePath) async {
```

Comment:
Instead of a `String localePath`, AssetLoader should receive `Locale locale`, so any path logic can be a custom implementation. This could also allow us to remove `useOnlyLangCode` parameter, and let the user implement this logic in an `AssetLoader`. Although it may make the lib _less easy_, as the user would need to implement `RootBundleAssetLoader` logic.

Alternatively, we can provide a `resolvePath` parameter, with the following signature:
```dart
Future<String> resolvePath({Locale locale, String basePath})
```
This way, we can deprecate `useOnlyLangCode` and the user can easily provide its own paths and AssetLoader would receive the generated `localePath`